### ISSUE TITLE
Default to current version of PHP

### ISSUE THREAD

**Issue by lornajane** [cloudfoundry/php-buildpack#233]
What version of Cloud Foundry and CF CLI are you using? (i.e. What is the output of running `cf curl /v2/info && cf version`?
`{"name":"Bluemix","build":"270013","support":"http://ibm.biz/bluemix-supportinfo","version":0,"description":"IBM Bluemix","authorization_endpoint":"https://login.eu-gb.bluemix.net/UAALoginServerWAR","token_endpoint":"https://uaa.eu-gb.bluemix.net","min_cli_version":null,"min_recommended_cli_version":null,"api_version":"2.92.0","app_ssh_endpoint":"ssh.eu-gb.bluemix.net:2222","app_ssh_host_key_fingerprint":"c7:1f:89:2a:62:3b:78:a9:08:c9:33:81:fb:39:26:da","app_ssh_oauth_client":"ssh-proxy","doppler_logging_endpoint":"wss://doppler.eu-gb.bluemix.net:443"}`

`cf version 6.32.0+0191c33d9.2017-09-26`

What version of the buildpack you are using?
4.3.39

If you were attempting to accomplish a task, what was it you were attempting to do?
Deploy a PHP app (actually a single `index.php` file)

What did you expect to happen?
The PHP app to be deployed with some current version of PHP (7.0 or 7.1)

What was the actual behavior?
App was deployed with PHP 5.6.31.  This is not a supported version of PHP (see http://php.net/supported-versions.php)

Please confirm where necessary:
* [ ] I have included a log output
* [ ] My log includes an error message
* [x] I have included steps for reproduction

This buildpack offers a very good selection of PHP versions and keeps up with everything brilliantly, thankyou!  I would however like to propose that the default PHP version always be a recommended stable one - so today that should be PHP 7.1.

**Comment by dmikusa-pivotal** [comment]
>App was deployed with PHP 5.6.31. This is not a supported version of PHP (see http://php.net/supported-versions.php)

The link says that it's supported for security fixes through the end of 2018.

>I would however like to propose that the default PHP version always be a recommended stable one - so today that should be PHP 7.1.

+1 for bumping the default version, with the appropriate fanfare & announcements of course.  

We have to be careful though.  Users that are on 5.6 now would essentially get promoted to the new version we pick (unless they've pinned the version to 5.6) the next time they stage, and jumping from 5.6 to 7.x is a big jump for apps.  I think we want to be careful as this could break user's apps.  I'm wondering if bumping to 7.0 would make more sense?  It's less of a jump, so it might be an easier transition?  It's also got security support for a while still, so maybe this would be a better path?  Then we could transition from 7.0 to 7.1 later in 2018.  Thoughts?

For what it's worth, the buildpack has historically been conservative in it's default selection.  It has often been in the middle of supported versions; not the oldest version of PHP we support, but not the newest either.  Jumping to 7.1 would mean we're flipping to be less conservative and more leading-edge.

**Comment by lornajane** [comment]
Thanks for the swift and thoughtful response!  I am hoping that since PHP 7.2 is likely to be released this week, going to PHP 7.1, which is a year old but gets active support, won't be too much like forcing users onto the bleeding edge where they might be at risk of unexpected effects.  I'd prefer to see a version with active support as the default choice for users who don't know which version they need, with the existing choices still available for anyone who needs a particular version.  Continued conservative progress is of course also fine, I am just lobbying for my own expectations as a developer to be met really :)

**Comment by sclevine** [comment]
PHP 7.2 was just released today. I support moving to 7.2, so that we can reduce the frequency of default version changes in the future.

This is breaking change in a buildpack, and will require 30, 60, or 90 days notice.

@dmikusa-pivotal, do you have an opinion about which time period makes sense here? I would suggest 30 days because we aren't going to remove 5.6 from the buildpack.

**Comment by dmikusa-pivotal** [comment]
@sclevine - Thoughts:

 - I'm not sure we should switch to the 7.2 as default.  That might be too aggressive.  I'd be more comfortable with 7.1 as default.  7.2 as latest & 5.6 as legacy.  I think we could even drop 7.0.  

- Longer notice would be better:  60 or 90 days.  

- If there's some way the buildpack could alert on this change that would be even better.  I know we do this if the buildpack version changes.  Maybe we could do something similar.  Keep track of the previously installed PHP version in the buildpack cache.  If that changes, the buildpack could then alert.  I think this would help for this particular case, but would also be generally useful as it would let people see when maintenance releases of PHP are changing too.

Dan

**Comment by cf-buildpacks-eng** [comment]
I really like the suggestion about the default version changing. @sclevine
should we start doing this across the board?

**Comment by noo-zh** [comment]
I would suggest the same like @dmikusa-pivotal says.

Question: when we could expect 7.2 in the buildpack? I'm looking forward to some of new features...

**Comment by lornajane** [comment]
Just revisiting this to check if there's anything I can do to help progress this?  There's some consensus here about moving to a current version as the default, which is great, but can I do more than just cheer on this excellent project? :)

**Comment by dgodd** [comment]
@lornajane You can always set `"PHP_VERSION": "{PHP_72_LATEST}"` in the `.bp-config/options.json` file.

**Comment by lornajane** [comment]
Yes, I understand how to use this myself.  But as more and more PHP projects move to expecting PHP 7, my question is whether we can move the default behaviour of the buildpack to set up more first-time users for success.  A default version is usually a safe choice and I felt like there was some good consensus around aiming for current stable as a default, allowing for a cautious route to that outcome so that people relying on the current default don't get nasty surprises!  Staying on PHP 5 by default is definitely a nasty surprise (in my personal opinion), especially as new projects deployed today are unlikely to be targeting it.

**Comment by dgodd** [comment]
Thanks for clarifying @lornajane .

As to how to progress this. We plan to do the update dance of sending an email to cf-dev, creating a git issue by the end of this week. Then it is just a matter of waiting the number of days

Thank you for your offer of help, but in this case it is just a matter of waiting for any arguments against
